blog/_posts/2021-01-19-scala-native-0.4-release.md

```diff
@@ -0,0 +1,225 @@
+post-type: blog
+by: Wojciech Mazur, Scala Center
+title: A sneak peek into the Scala Native 0.4 release
+---
+
+If you ever thought about running your perfect Scala code in an environment other than JVM, you've probably already met
+Scala Native---a project with a promise of instant startup, low memory usage, interop with C libraries, and competitive performance
+thanks to [Interflow optimizer](https://scala-native.readthedocs.io/en/latest/blog/interflow.html).
+
+However, since there have been no release in over a year, you may ask yourself---is this project even alive?<br>
+With the long-awaited release 0.4.0 it proves it is! 
+Today we are going to discuss the most impacting new features and changes coming in this new major release.
+We will start with changes according to Scala Native compiler plugin, especially supported Scala versions and cross-platform testing.
+Next we will talk about changes to C interoperability and SBT build that need to be taken into account since when working this release.
+Last, but not least we will discuss why we needed to break binary compatibility with previous versions.
+
+## Support for latest Scala 2 versions
+Until now Scala Native supported only Scala 2.11, which is now perceived by most of the developers as deprecated.
+We can proudly announce that Scala Native is now compatible with Scala versions 2.12.13 and 2.13.4, in addition to the 2.11.12 with known bugs fixed.
+
+## Reflective instantiation
+The SN plugin removes unused definitions of classes, methods, and variables during the linking stage to reduce the size of the output binary.
+This used to make using reflective calls very difficult. Fortunately, this is a thing of the past---with release 0.4.0 we
+introduced a subset of reflective operations allowed to be used.
+
+It would not be wise to enable reflective calls to all definitions as we would need to load many more objects and increase the size of the binary.
+That’s why you need to signal to the compiler which classes can use this mechanism - it can be done by marking your `trait`, `class`, or `object`
+with the `@EnableReflectiveInstantation` annotation. It will also make all of its descendants able to be reflectively instantiated.
+
+Annotated classes and modules, having a concrete implementation, can be accessed via the provided `scalanative.reflect.Reflect` API.
+If you have used Scala.js before, it may seem similar to you, as the new implementation uses exactly the same API.
+
+## Cross-platform testing using JUnit
+Until recently, all our internal tests were executed using our own native testing framework, mainly due to lack of proper
+reflection support in Scala Native. After getting rid of this blocker, we have decided to use JUnit instead and in 
+result implemented the native plugin for this framework. Thanks to this change you are also able to also use it in your own projects.  
+Naturally you are still able to use any other testing framework, as long as it supports Scala Native.
```

Comment:
Do such frameworks exist? If so, what are they? (Do popular Scala testing frameworks have, at least, about adding this support?)